src/ng/directive/ngOptions.js

```diff
@@ -303,6 +303,13 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
         values = values || [];
 
         Object.keys(values).forEach(function getWatchable(key) {
+          if (isString(values[key]) &&
```

Comment:
If it hasn't been monkey-patched, Object.keys() only produces a list of strings strings (even numeric indexes are strings).

On this point, we should probably make adjustments so that we always use the original un-monkey-patched Object.keys function